libcodechecker/analyze/store_handler.py

```diff
@@ -343,44 +343,76 @@ def addReport(storage_session,
         # large databases?
         reports = session.query(Report) \
             .filter(Report.bug_id == bug_hash, Report.run_id == run_id)
-        # Check for duplicates by bug hash.
-        LOG.debug("checking duplicates")
+
+        # Detection status for the newly stored reports.
+        new_report_status = 'new'
+
+        # Report id for a duplicate report, if it was already stored in
+        # the database. Bug hash and report path should match.
+        duplicate_report_id = None
```

Comment:
In python, you do not need to declare this upfront I think.

Reply:
Sometimes you have to. If the variable is only assigned in a conditional statement later on, and then is used in a "None-ness check" (as it is in this case), Python can end up reporting `local variable referenced before assigned`.